tools/canonicalize-proto-file/Main.hs

```diff
@@ -30,9 +30,54 @@ data Args w = Args
   { includeDir :: w ::: [FilePath] <?> "Path to search for included .proto files (can be repeated, and paths will be searched in order; the current directory is used if this option is not provided)"
   , proto      :: w ::: FilePath   <?> "Path to input .proto file"
   } deriving Generic
+
 instance ParseRecord (Args Wrapped)
+
 deriving instance Show (Args Unwrapped)
 
+-- | @('isOverlappingIntervals' a b :: 'Bool')@ is a relation between two 
+-- interval-like types @a@ and @b@ that is 'True' when the two intervals are 
+-- overlapping or touching over a subinterval of @a@ and @b@.
+isOverlappingIntervals :: (Ord a, Enum a) => (a, a) -> (a, a) -> Bool
+isOverlappingIntervals (x, y) (u, v) = y >= u || v <= x
```

Comment:
Is this a problem, or is there an extra precondition somewhere?
```
ghci> let isOverlappingIntervals (x, y) (u, v) = y >= u || v <= x 
ghci> isOverlappingIntervals (2, 3) (0, 1) 
True
```

Reply:
Thanks, I have fixed this.